### ISSUE TITLE
Could not retrieve pressure levels for ua with axes []

### ISSUE THREAD

**Issue by oloapinivad** [goord/ece2cmor3#144]
Hi all,
I am using master branch but working with the PRIMAVERA/CMIP6 tables from Jon Seddon  https://github.com/jonseddon/cmip6-cmor-tables, so combining them with parameters tables (ifspar and nemopar) from the PRIMAVERA branch. 

In this configuration, I am getting this error for all ua- and va-related variables

``````
ERROR:ece2cmor3.postproc:Could not retrieve pressure levels for ua with axes []
ERROR:ece2cmor3.cdoapi:(returncode:1) cdo showltype: Started child process "selcode,132 /marconi_scratch/userexternal/pdavini0/tmp_cmor/qctr_4998/tmp_1950_1/cmorized/CMIP6/132.128.100 (pipe1.1)".
Warning (cgribexScanTimestep2) : Record  9 (id=132.128 lev1=82500 lev2=0) timestep 2: Parameter not defined at timestep 1!
cdo(2) selcode: Open failed on >/marconi_scratch/userexternal/pdavini0/tmp_cmor/qctr_4998/tmp_1950_1/cmorized/CMIP6/132.128.100<
Unsupported file structure
``````

I guess this is a problem related to the adoption of Jon Seddon's tables, since this is not happening when I use the official CMIP6 tables (the one that comes together with the master branch). However, inspection of the tables does not show any significant difference. 
Do you have any suggestions? 
I am getting this "unsupported file structure" for ua and va families (ua,ua27,ua7h etc.) but not for zg or other pressure levels vars.
Thanks again!

**Comment by goord** [comment]
Hi Paolo, this must be due to the extra Primavera output at 850 hPa, which is in the spectral file at 3-hourly basis and probably not filtered out appropriately. Can you post the output of
```
grib_ls -w,paramId=132,levelType=100,dataDate=19500101 -p,paramId,levelType,level,dataDate,dataTime /marconi_scratch/userexternal/pdavini0/tmp_cmor/qctr_4998/tmp_1950_1/cmorized/CMIP6/132.128.100
```

I bet the 850hPa level is still in there every 3 hours...

**Comment by oloapinivad** [comment]
Hi @goord,

thanks for the reply. 
You are right, the wind at 850hPa is still there at time 0300 and 0900 (3hourly) and it is doubled at 0000 and 0600. 
Here below the output of the command:

`````
paramId     levelType   level       dataDate    dataTime    
132         pl          1000        19500101    0          
132         pl          975         19500101    0          
132         pl          950         19500101    0          
132         pl          925         19500101    0          
132         pl          900         19500101    0          
132         pl          875         19500101    0          
132         pl          850         19500101    0          
132         pl          850         19500101    0          
132         pl          825         19500101    0          
132         pl          800         19500101    0          
132         pl          775         19500101    0          
132         pl          750         19500101    0          
132         pl          700         19500101    0          
132         pl          650         19500101    0          
132         pl          600         19500101    0          
132         pl          550         19500101    0          
132         pl          500         19500101    0          
132         pl          450         19500101    0          
132         pl          400         19500101    0          
132         pl          350         19500101    0          
132         pl          300         19500101    0          
132         pl          250         19500101    0          
132         pl          225         19500101    0          
132         pl          200         19500101    0          
132         pl          175         19500101    0          
132         pl          150         19500101    0          
132         pl          125         19500101    0          
132         pl          100         19500101    0          
132         pl          70          19500101    0          
132         pl          50          19500101    0          
132         pl          30          19500101    0          
132         pl          20          19500101    0          
132         pl          10          19500101    0          
132         pl          5           19500101    0          
132         pl          1           19500101    0          
132         pl          850         19500101    300        
132         pl          1000        19500101    600        
132         pl          975         19500101    600        
132         pl          950         19500101    600        
132         pl          925         19500101    600        
132         pl          900         19500101    600        
132         pl          875         19500101    600        
132         pl          850         19500101    600        
132         pl          850         19500101    600        
132         pl          825         19500101    600        
132         pl          800         19500101    600        
132         pl          775         19500101    600        
132         pl          750         19500101    600        
132         pl          700         19500101    600        
132         pl          650         19500101    600        
132         pl          600         19500101    600        
132         pl          550         19500101    600        
132         pl          500         19500101    600        
132         pl          450         19500101    600        
132         pl          400         19500101    600        
132         pl          350         19500101    600        
132         pl          300         19500101    600        
132         pl          250         19500101    600        
132         pl          225         19500101    600        
132         pl          200         19500101    600        
132         pl          175         19500101    600        
132         pl          150         19500101    600        
132         pl          125         19500101    600        
132         pl          100         19500101    600        
132         pl          70          19500101    600        
132         pl          50          19500101    600        
132         pl          30          19500101    600        
132         pl          20          19500101    600        
132         pl          10          19500101    600        
132         pl          5           19500101    600        
132         pl          1           19500101    600        
132         pl          850         19500101    900       
`````

**Comment by goord** [comment]
Hi Paolo, I am having difficulty reproducing your issue. Could you make sure you run the latest version and post the request variable list that leads to this error? thanks

**Comment by oloapinivad** [comment]
Hi @goord here attached the varlist I am using (converted in txt to upload it)
[varlist-cmip6-paolo.txt](https://github.com/goord/ece2cmor3/files/1856420/varlist-cmip6-paolo.txt)

I have just tried with the last version, again the same issue (I am using a PRIMAVERA-produced file, do you need it also?). Here below the full log!
Let me know if I can help you more.
Thanks,
P

`````
+ /marconi_work/Pra13_3311/ecearth3/PRIMAVERA/cmorize/ece2cmor3/ece2cmor3/ece2cmor.py /marconi_scratch/userexternal/pdavini0/tmp_cmor/qctr_30924/tmp_1950_1/linkdata 1950-01-01 --exp qctr --conf /marconi_scratch/userexternal/pdavini0/tmp_cmor/qctr_30924/tmp_1950_1/cmorized/metadata-qctr-leg000.json --vars /marconi/home/userexternal/pdavini0/ecearth3/ece3-postproc/ece2cmor3_support/varlist/varlist-cmip6-paolo.json --npp 32 --tmpdir /marconi_scratch/userexternal/pdavini0/tmp_cmor/qctr_30924/tmp_1950_1/cmorized --tabid CMIP6 --tabdir /marconi_work/Pra13_3311/ecearth3/PRIMAVERA/cmorize/cmip6-cmor-tables/Tables --mode append --atm --filter
ERROR:ece2cmor3.taskloader:The siprn variable does not appear in the CMOR table file SImon
ERROR:ece2cmor3.taskloader:The precip variable does not appear in the CMOR table file Omon
ERROR:ece2cmor3.taskloader:The sistrxdtop variable does not appear in the CMOR table file SIday
ERROR:ece2cmor3.taskloader:The sistrydtop variable does not appear in the CMOR table file SIday
ERROR:ece2cmor3.taskloader:The orog variable does not appear in the CMOR table file AERmon
ERROR:ece2cmor3.taskloader:The psl variable does not appear in the CMOR table file E3hrPt
ERROR:ece2cmor3.taskloader:The snw variable does not appear in the CMOR table file Eday
ERROR:ece2cmor3.taskloader:The mrro variable does not appear in the CMOR table file Eday
INFO:ece2cmor3.taskloader:Found 229 cmor target variables in input variable list.
ERROR:ece2cmor3.taskloader:Could not find parameter table entry for tso in table 3hr...skipping variable. This variable is missing
INFO:ece2cmor3.taskloader:Created 170 ece2cmor tasks from input variable list.
INFO:ece2cmor3.ece2cmorlib:Selected 170 IFS tasks from 170 input tasks
INFO:ece2cmor3.grib_filter:Searching for previous month file of /marconi_scratch/userexternal/pdavini0/tmp_cmor/qctr_30924/tmp_1950_1/linkdata/ICMGGqctr+195001
INFO:ece2cmor3.ifs2cmor:Inspecting 170 tasks.
INFO:ece2cmor3.ifs2cmor:Validated 170 tasks for processing.
INFO:ece2cmor3.ifs2cmor:Executing 170 IFS tasks...
INFO:ece2cmor3.ifs2cmor:Post-processing tasks...
INFO:ece2cmor3.ifs2cmor:Post-processing 171 IFS tasks...
ERROR:ece2cmor3.cdoapi:(returncode:1) cdo showltype: Started child process "selcode,131 /marconi_scratch/userexternal/pdavini0/tmp_cmor/qctr_30924/tmp_1950_1/cmorized/131.128.210 (pipe1.1)".
Warning (cgribexScanTimestep2) : Record  9 (id=131.128 lev1=82500 lev2=0) timestep 2: Parameter not defined at timestep 1!
cdo(2) selcode: Open failed on >/marconi_scratch/userexternal/pdavini0/tmp_cmor/qctr_30924/tmp_1950_1/cmorized/131.128.210<
Unsupported file structure

ERROR:ece2cmor3.postproc:Could not retrieve pressure levels for ua with axes []
ERROR:ece2cmor3.cdoapi:(returncode:1) cdo showltype: Started child process "selcode,132 /marconi_scratch/userexternal/pdavini0/tmp_cmor/qctr_30924/tmp_1950_1/cmorized/132.128.210 (pipe1.1)".
Warning (cgribexScanTimestep2) : Record  9 (id=132.128 lev1=82500 lev2=0) timestep 2: Parameter not defined at timestep 1!
cdo(2) selcode: Open failed on >/marconi_scratch/userexternal/pdavini0/tmp_cmor/qctr_30924/tmp_1950_1/cmorized/132.128.210<
Unsupported file structure
``````

**Comment by goord** [comment]
Ah yes I see it now, the combination of ua850 in E3hrPt and ua7h in 6hrPlevPt breaks the filtering...

**Comment by goord** [comment]
Should be fixed after latest commit 2540e793f8139bbc50d2f950505d139f536c142e

**Comment by oloapinivad** [comment]
Great, thanks @goord, now it works!
